lib/Threading/Pthreads.cpp

```diff
@@ -101,18 +101,26 @@ swift::threading_impl::thread_get_current_stack_bounds() {
 #if defined(__FreeBSD__)
   if (pthread_attr_init(&attr))
     return {};
-#endif
 
-  if (!pthread_getattr_np(pthread_self(), &attr)) {
-    if (!pthread_attr_getstack(&attr, &begin, &size)) {
-      stack_bounds result = { begin, (char *)begin + size };
-      pthread_attr_destroy(&attr);
-      return result;
-    }
+  if (pthread_attr_get_np(pthread_self(), &attr)) {
+    pthread_attr_destroy(&attr);
+    return {};
+  }
+#elif defined(__linux__)
+  if (pthread_getattr_np(pthread_self(), &attr))
+    return {};
```

Comment:
Missing a `pthread_attr_destroy(&attr);`

Reply:
`pthread_getattr_np()` calls `pthread_attr_destroy()` itself on error (see source [here](https://sourceware.org/git?p=glibc.git;a=blob;f=nptl/pthread_getattr_np.c;h=9c5b73b452ab0faa70f8e734c6c725e98f2fa7e3;hb=HEAD#l204)), so I don't think we need to (nor should) call it ourselves.

The only other error return in `pthread_getattr_np()` is when its own `pthread_attr_init()` fails, which also means that we don't need to call destroy.